lib/Monitoring/Plugin.pm

```diff
@@ -163,6 +156,10 @@ sub add_arg {
 sub getopts {
     my $self = shift;
 	$self->opts->getopts(@_) if $self->_check_for_opts;
+	$self->set_thresholds(
+		warning  => $self->opts->warning,
+		critical => $self->opts->critical,
+	) if ( defined $self->opts->warning && defined $self->opts->critical );
```

Comment:
Ok, but what if only one of the threshold is set (warning or critical)? Maybe I'm mistaken but I though that's a valid/common use case (maybe you want to only warn, never go critical...)

Reply:
thats a valid point. It previously checked $self->opts while it now checks critical and warning. This should at least be an `or` or reverted to the opts check.

Reply:
Yeah sure, we'll switch it to an `or`

Reply:
@dermoth @sni fix: https://github.com/monitoring-plugins/monitoring-plugin-perl/pull/23